### ISSUE TITLE
improve ctest handling

### ISSUE THREAD

**Issue by nekopsykose** [chimera-linux/cports#236]
currently, cmake tests are ran via `ninja`/`make` `test`. this sucks because:

- it does not actually run anything in parallel (`ctest -j`/`CTEST_PARALLEL_LEVEL=`), which works for ~80% of the test suites out there to make them a lot faster
- `CTEST_OUTPUT_ON_FAILURE`/`--output-on-failure` is not set so the failures don't log anything and just say "test failed", which is meaningless by itself

**Comment by nekopsykose** [comment]
would also be nice to have a way to pass the `-E ..` for test skips without needing to override do_check. if ctest was the actual invoked command then the make_check_args probably just work..
